core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -447,7 +447,15 @@ public String getServiceHost(String serviceName, Integer servicePort) {
      * @return a port that can be used for accessing the service container.
      */
     public Integer getServicePort(String serviceName, Integer servicePort) {
-        return ambassadorContainer.getMappedPort(ambassadorPortMappings.get(getServiceInstanceName(serviceName)).get(servicePort));
+        Map<Integer, Integer> portMap = ambassadorPortMappings.get(getServiceInstanceName(serviceName));
+
+        if(portMap == null) {
+            throw new IllegalArgumentException("\n\nContext: Getting the port of '" + serviceName + "'.\n" +
```

Comment:
I feel a bit uncomfortable with newlines in exception messages, especially at the beginning. Some proposed rewording:

```suggestion
            throw new IllegalArgumentException("Could not get a port for '" + serviceName + "'. " +
```